infra/Pos/Slotting/MemState/Holder.hs

```diff
@@ -46,8 +45,18 @@ askSlottingTimestamp  = fst <$> askSlotting
 -- MonadSlotsData implementation
 ----------------------------------------------------------------------------
 
-instance MonadIO m =>
-         MonadSlotsData (SlottingHolder m) where
+data SlotsDataRedirectTag
+
+type SlotsDataRedirect =
+    Ether.TaggedTrans SlotsDataRedirectTag IdentityT
+
+runSlotsDataRedirect :: SlotsDataRedirect m a -> m a
+runSlotsDataRedirect = coerce
```

Comment:
I'm not sure where to put haddock descriptions, this is a common pattern throughout the request: an `IdentityT` wrapper used to redirect a class implementation to something beneath it.

What `SlotsDataRedirect` does when added to a monad transformer stack is that it adds an instance — `MonadSlotsData` — that works using the underlying `MonadSlotting m`.

It would be possible to write an instance like this:

```
(MonadSlotting m, MonadIO m) => MonadSlotsData m
```

but we need the `IdentityT` wrapper for two reasons:

* to avoid orphan instances. They are bad, especially when combined with `{-# OVERLAPPABLE #-}` instances. Coherency violations are a nightmare to debug.
* to make it possible to have multiple independent implementations without conflicts. For example, `BalancesRedirect` and `BalancesWalletRedirect` are different redirection transformers.